spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -569,34 +450,29 @@ public Object doWithRetry(RetryContext context) throws RuntimeException {
 			listenedPartitions = new ArrayList<Partition>();
 			for (Partition partition : allPartitions) {
 				// divide partitions across modules
-				if (accessor.getPartitionIndex() != -1) {
-					if ((partition.getId() % moduleCount) == accessor.getPartitionIndex()) {
+				if (properties.isPartitioned()) {
+					if ((partition.getId() % moduleCount) == properties.getPartitionIndex()) {
 						listenedPartitions.add(partition);
 					}
 				}
 				else {
-					int moduleSequence = accessor.getSequence();
-					if (moduleCount == 0) {
-						throw new IllegalArgumentException("The Kafka transport does not support 0-count modules");
-					}
-					else {
-						// sequence numbers are zero-based
-						if ((partition.getId() % moduleCount) == (moduleSequence - 1)) {
-							listenedPartitions.add(partition);
-						}
+					int instanceIndex = properties.getInstanceIndex();
+					// sequence numbers are zero-based
+					if ((partition.getId() % moduleCount) == instanceIndex) {
+						listenedPartitions.add(partition);
 					}
 				}
 			}
 		}
 		topicsInUse.put(name, listenedPartitions);
-		ReceivingHandler rh = new ReceivingHandler();
+		ReceivingHandler rh = new ReceivingHandler(properties);
 		rh.setOutputChannel(moduleInputChannel);
 
 		final FixedSubscriberChannel bridge = new FixedSubscriberChannel(rh);
 		bridge.setBeanName("bridge." + name);
 
 		final KafkaMessageListenerContainer messageListenerContainer =
-				createMessageListenerContainer(accessor, group, maxConcurrency, listenedPartitions,
+				createMessageListenerContainer(properties, group, properties.getConcurrency(), listenedPartitions,
```

Comment:
probably don't need the separate concurrency param since that is included in the properties